pkg/pdp/service/piece_upload.go

```diff
@@ -23,7 +24,16 @@ import (
 	"github.com/storacha/piri/pkg/pdp/types"
 )
 
-func (p *PDPService) UploadPiece(ctx context.Context, pieceUpload types.PieceUpload) error {
+func (p *PDPService) UploadPiece(ctx context.Context, pieceUpload types.PieceUpload) (retErr error) {
+	start := time.Now()
+	log.Infow("uploading piece", "request", pieceUpload)
+	defer func() {
+		if retErr != nil {
```

Comment:
I guess this will be a thing then :P

Question: all these logs are unrelated to the single process refactor yes? generally I would put those in a seperate PR. Not a big deal though.